### DISCUSSION TITLE
JPG extension choice equivalent for DNG files

### DISCUSSION THREAD

**Discussion by gyansinha** [RhetTbull/osxphotos#1640] [Ideas]
Is there an equivalent to the "jpeg_ext" option for DNG files, like "dng_ext"? I ask because it looks like Photos converts the extension to lowercase (or maybe Leica Fotos does) and that is what gets exported - I'd like to preserve the original filename AND extension during the export.

**Comment by RhetTbull** [comment]
No there is no such option. OSXPhotos should preserve the case of the extension as it appears in the Photos library. I just did  test and created 2 DNG files, one `fizz.dng` and `buzz.DNG` and imported both then used OSXPhotos to export them and the case was preserved.

There is an open issue (#714) to add options to force the case of the extension to upper or lower as some people have a preference for one or the other.

As an alternative in the meantime, you could do something like this using the `--post-command` option to change the extension. This works because OSXPhotos assumes the target export volume is case-insensitive as most Mac volumes are and thus compares filenames using case-insensitive comparison. Thus, if you exported `foo.dng` and renamed it `foo.DNG`, when you export again with `--update`, these appear to be the same file to OSXPhotos which allows the export logic to correctly decide whether or not the file needs to be updated.

`--post-command exported "mv {filepath|shell_quote} {filepath.parent|shell_quote}/{filepath.stem|shell_quote}{filepath.suffix|upper}"`

This uses the post-command option to run a command on all exported files that uses `mv` to rename the file to `filepath/filename.EXT` format. The `|shell_quote` is needed properly quote / escape any spaces or special characters in the path name. The file bit `{filepath.suffix|upper}` uses the `|upper` template filter to change the suffix (extension) to uppercase when re-constructing the target filepath.

**Reply by gyansinha** [reply]
Thanks - I'll try what you suggest. I've been using a TOML config file, any ideas about how to incorporate it there?

**Reply by RhetTbull** [reply]
> I've been using a TOML config file, any ideas about how to incorporate it there?

Add this to the toml file under the `[export]` section:

`post_command = [ [ "exported", "mv {filepath|shell_quote} {filepath.parent|shell_quote}/{filepath.stem|shell_quote}{filepath.suffix|upper}",],]`

**Reply by gyansinha** [reply]
Thanks - that worked like a charm! Appreciate your work on this.
